ext/standard/pack.c

```diff
@@ -1003,108 +989,78 @@ PHP_FUNCTION(unpack)
 						break;
 					}
 
-					case 'c':
-					case 'C': {
-						int issigned = (type == 'c') ? (input[inputpos] & 0x80) : 0;
-						zend_long v = php_unpack(&input[inputpos], 1, issigned, byte_map);
+					case 'c':   /* signed */
+					case 'C': { /* unsigned */
+						uint8_t x = input[inputpos];
+						zend_long v = (type == 'c') ? (int8_t) x : x;
 						add_assoc_long(return_value, n, v);
 						break;
 					}
 
-					case 's':
-					case 'S':
-					case 'n':
-					case 'v': {
-						zend_long v;
-						int issigned = 0;
-						int *map = machine_endian_short_map;
+					case 's':   /* signed machine endian   */
+					case 'S':   /* unsigned machine endian */
+					case 'n':   /* unsigned big endian     */
+					case 'v': { /* unsigned little endian  */
+						zend_long v = 0;
+						uint16_t x;
+						memcpy(&x, &input[inputpos], 2);
```

Comment:
Might be nicer to load through `unaligned_uint16_t` instead? (Grep for existing typedefs...)

Reply:
You mean like this: uint16_t x = *((unaligned_uint16_t*) &input[inputpos])

Isn't in C access to char array through pointer of any other type forbidden?

Reply:
We don't really care about strict aliasing as long as it's not actively miscompiled ;) I'm okay with keeping the memcpy though, I expect it to compile down to about the same.

Reply:
Ok then, I'll change it.